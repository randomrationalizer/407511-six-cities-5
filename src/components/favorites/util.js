export const getCitiesFromOffers = (offers) => {
  const cities = offers.slice().map((offer) => offer.city.name).sort();
  return [...new Set(cities)];
};
