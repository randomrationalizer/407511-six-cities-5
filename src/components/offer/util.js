import dayjs from "dayjs";
import {OfferType} from "../../const";


export const capitalize = (str) => {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

// Возвращает дату в виде строки в формате: 2019-04-24
export const formatDate = (date) => {
  return dayjs(date).format(`YYYY-MM-DD`);
};

// Возвращает дату в виде строки в формате: January 2020
export const humanizeDate = (date) => {
  return dayjs(date).format(`MMMM YYYY`);
};

// Сравнение для сортировки отзывов в хронологическом порядке
export const sortByDate = (a, b) => {
  return dayjs(a.date) - dayjs(b.date);
};

export const getDescriptionSentences = (text) => {
  let description = text.split(`. `);
  description[description.length - 1] = description[description.length - 1].replace(`.`, ``);

  return description;
};

export const isFavoritesCard = (type) => {
  return type === OfferType.FAVORITES;
};

export const isMainPageCard = (type) => {
  return type === OfferType.MAIN;
};
