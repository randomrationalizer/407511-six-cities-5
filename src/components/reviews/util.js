const REVIEW_MIN_LENGTH = 50;
const REVIEW_MAX_LENGTH = 300;
const RATING_MIN = 1;
const RATING_MAX = 5;

export const checkCommentValidity = (comment) => {
  return comment.length >= REVIEW_MIN_LENGTH && comment.length <= REVIEW_MAX_LENGTH ? true : false;
};

export const checkRatingValidity = (rating) => {
  return rating >= RATING_MIN && rating <= RATING_MAX ? true : false;
};
