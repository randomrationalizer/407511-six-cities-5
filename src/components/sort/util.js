import {SortType} from "../../const";

export const sortTypeToTitle = {
  [SortType.DEFAULT]: `Popular`,
  [SortType.PRICE_UP]: `Price: low to high`,
  [SortType.PRICE_DOWN]: `Price: high to low`,
  [SortType.RATING_DOWN]: `Top rated first`
};
