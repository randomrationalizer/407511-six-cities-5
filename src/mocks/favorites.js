export const favorites = [`1`, `3`, `4`];
