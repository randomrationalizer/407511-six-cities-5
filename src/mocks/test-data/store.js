import configureStore from "redux-mock-store";
import {NameSpace} from "../../store/root-reducer";
import mockOffers from "./offers";
import mockReviews from "./reviews";
import mockNearbyOffers from "./nearby-offers";
import mockCurrentOffer from "./current-offer";
import mockUserInfo from "./user-info";
import mockFavorites from "./favorites";
import mockCities from "./cities";
import {SortType, AuthorizationStatus, DEFAULT_CITY} from "../../const";


export default {
  [NameSpace.APP_DATA]: {
    currentCity: DEFAULT_CITY,
    allOffers: mockOffers,
    cities: mockCities,
    currentSort: SortType.DEFAULT
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
    isAuthRequestComplete: true,
    userInfo: mockUserInfo,
    favorites: mockFavorites
  },
  [NameSpace.CURRENT_OFFER]: {
    offer: mockCurrentOffer,
    reviews: mockReviews,
    nearbyOffers: mockNearbyOffers
  },
  [NameSpace.LOAD_STATUS]: {
    isOffersLoaded: true,
    isFavoritesLoaded: true,
    isCurrentOfferLoaded: true,
    errorMessage: null,
    isLoadFinished: true
  }
};
