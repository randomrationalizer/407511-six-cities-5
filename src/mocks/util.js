import configureStore from "redux-mock-store";

export const noop = () => {};

export const getMockStore = (store) => {
  const storeConfig = configureStore();
  return storeConfig(store);
};

export const nope = () => {
  return Promise.resolve({});
};
