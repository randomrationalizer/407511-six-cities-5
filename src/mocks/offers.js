import {getPhoto, generateDescription} from "./util.js";


export const offers = [
  {
    id: `1`,
    title: `Beautiful & luxurious apartment at great location `,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
    price: 6000,
    type: `apartment`,
    description: generateDescription(),
    bedrooms: 3,
    guests: 4,
    rating: 4.8,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Wi-Fi`, `Washing machine`, `Towels`],
    owner: {
      name: `Angelina`,
      avatar: getPhoto(),
      isHighRated: true
    },
    isPremial: true
  },

  {
    id: `2`,
    title: `Nice, cozy, warm big bed room`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
    price: 20,
    type: `room`,
    description: generateDescription(),
    bedrooms: 1,
    guests: 1,
    rating: 3.4,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`],
    owner: {
      name: `Mike`,
      avatar: getPhoto(),
      isHighRated: false
    },
    isPremial: false
  },

  {
    id: `3`,
    title: `Canal View Prinsengracht`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
    price: 200,
    type: `hotel`,
    description: generateDescription(),
    bedrooms: 2,
    guests: 3,
    rating: 4.1,
    photos: [
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    owner: {
      name: `Bob`,
      avatar: getPhoto(),
      isHighRated: true
    },
    isPremial: false
  },

  {
    id: `4`,
    title: ` Wood and stone place`,
    city: `Amsterdam`,
    coords: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
    price: 1000,
    type: `house`,
    description: generateDescription(),
    bedrooms: 6,
    guests: 7,
    rating: 3.7,
    photos: [
      {
        src: getPhoto(),
        description: ``
      },
      {
        src: getPhoto(),
        description: ``
      }
    ],
    options: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    owner: {
      name: `James`,
      avatar: getPhoto(),
      isHighRated: false
    },
    isPremial: false
  }
];
