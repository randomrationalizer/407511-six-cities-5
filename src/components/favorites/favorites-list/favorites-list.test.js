import React from "react";
import {Provider} from 'react-redux';
import {BrowserRouter} from "react-router-dom";
import renderer from "react-test-renderer";
import FavoritesList from "./favorites-list";
import store from "../../../mocks/test-data/store";
import mockFavorites from "../../../mocks/test-data/favorites";
import {getMockStore} from "../../../mocks/util";


const mockStore = getMockStore(store);


it(`Should FavoritesList renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <BrowserRouter>
            <FavoritesList
              favorites={mockFavorites}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

