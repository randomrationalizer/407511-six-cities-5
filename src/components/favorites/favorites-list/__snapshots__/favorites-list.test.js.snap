// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesList renders correctly 1`] = `
<main
  className="page__main page__main--favorites"
>
  <div
    className="page__favorites-container container"
  >
    <section
      className="favorites"
    >
      <h1
        className="favorites__title"
      >
        Saved listing
      </h1>
      <ul
        className="favorites__list"
      >
        <li
          className="favorites__locations-items"
        >
          <div
            className="favorites__locations locations locations--current"
          >
            <div
              className="locations__item"
            >
              <a
                className="locations__item-link"
                href="/#cologne"
                onClick={[Function]}
              >
                <span>
                  Cologne
                </span>
              </a>
            </div>
          </div>
          <div
            className="favorites__places"
          >
            <article
              className="favorites__card place-card card-favorites"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/hotels/69"
                  onClick={[Function]}
                >
                  <img
                    alt="Penthouse, 4-5 rooms + 5 balconies"
                    className="place-card__image"
                    height={110}
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg"
                    width={150}
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      228
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button  button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/hotels/69"
                    onClick={[Function]}
                  >
                    Penthouse, 4-5 rooms + 5 balconies
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Room
                </p>
              </div>
            </article>
          </div>
        </li>
        <li
          className="favorites__locations-items"
        >
          <div
            className="favorites__locations locations locations--current"
          >
            <div
              className="locations__item"
            >
              <a
                className="locations__item-link"
                href="/#hamburg"
                onClick={[Function]}
              >
                <span>
                  Hamburg
                </span>
              </a>
            </div>
          </div>
          <div
            className="favorites__places"
          >
            <article
              className="favorites__card place-card card-favorites"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/hotels/67"
                  onClick={[Function]}
                >
                  <img
                    alt="Penthouse, 4-5 rooms + 5 balconies"
                    className="place-card__image"
                    height={110}
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
                    width={150}
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      190
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button  button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "62%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/hotels/67"
                    onClick={[Function]}
                  >
                    Penthouse, 4-5 rooms + 5 balconies
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Room
                </p>
              </div>
            </article>
            <article
              className="favorites__card place-card card-favorites"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/hotels/68"
                  onClick={[Function]}
                >
                  <img
                    alt="Loft Studio in the Central Area"
                    className="place-card__image"
                    height={110}
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg"
                    width={150}
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      301
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button  button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "70%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/hotels/68"
                    onClick={[Function]}
                  >
                    Loft Studio in the Central Area
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
          </div>
        </li>
        <li
          className="favorites__locations-items"
        >
          <div
            className="favorites__locations locations locations--current"
          >
            <div
              className="locations__item"
            >
              <a
                className="locations__item-link"
                href="/#paris"
                onClick={[Function]}
              >
                <span>
                  Paris
                </span>
              </a>
            </div>
          </div>
          <div
            className="favorites__places"
          >
            <article
              className="favorites__card place-card card-favorites"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="favorites__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/hotels/66"
                  onClick={[Function]}
                >
                  <img
                    alt="The Pondhouse - A Magical Place"
                    className="place-card__image"
                    height={110}
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg"
                    width={150}
                  />
                </a>
              </div>
              <div
                className="favorites__card-info place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      111
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button  button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height={19}
                      width={18}
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "44%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/hotels/66"
                    onClick={[Function]}
                  >
                    The Pondhouse - A Magical Place
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  House
                </p>
              </div>
            </article>
          </div>
        </li>
      </ul>
    </section>
  </div>
</main>
`;
